pkg/clusteragent/admission/mutate/autoinstrumentation/auto_instrumentation.go

```diff
@@ -590,18 +587,122 @@ func (w *Webhook) newInjector(startTime time.Time, pod *corev1.Pod, opts ...inje
 		podMutator(w.version)
 }
 
+// podSumRessourceRequirements computes the sum of cpu/memory necessary for the whole pod.
+// This is computed as max(max(initContainer resources), sum(container resources)) for both limit and request
+func podSumRessourceRequirements(pod *corev1.Pod) corev1.ResourceRequirements {
+	ressourceRequirement := corev1.ResourceRequirements{
+		Limits:   corev1.ResourceList{},
+		Requests: corev1.ResourceList{},
+	}
+
+	for _, k := range [2]corev1.ResourceName{corev1.ResourceMemory, corev1.ResourceCPU} {
+		// Take max(initContainer ressource)
+		for i := range pod.Spec.InitContainers {
+			c := &pod.Spec.InitContainers[i]
+			if limit, ok := c.Resources.Limits[k]; ok {
+				existing := ressourceRequirement.Limits[k]
+				if limit.Cmp(existing) == 1 {
+					ressourceRequirement.Limits[k] = limit
+				}
+			}
+			if request, ok := c.Resources.Requests[k]; ok {
+				existing := ressourceRequirement.Requests[k]
+				if request.Cmp(existing) == 1 {
+					ressourceRequirement.Requests[k] = request
+				}
+			}
```

Comment:
I still think that it is better to not set a different value between request and limit because we want to limit the impact on other application that might run on the same node.